### DISCUSSION TITLE
Trouble with Odin

### DISCUSSION THREAD

**Discussion by TheDakk** [mbround18/valheim-docker#112] [Help!] [answered]
Trying to get Odin to run:

I installed Rust under the root account, then switched over to the docker account and ran:

cargo install --git https://github.com/mbround18/valheim-docker.git --branch main

docker run -it --name=steamcmd cm2network/steamcmd bash

If I CD to /vaheim/server, I can stop the server but when I start the server I get the following error in the server log:

Error, unable to resolve user name to user id

Hopefully I didn't mess anything up! lol

EDIT: I guess the docker command I ran only went into the remote shell and didn’t do anything?

So I restarted from scratch, installed Rust and SteamCMD, but now running into this:

EDIT 2: $ cargo install --git https://github.com/mbround18/valheim-docker.git --branch main
    Updating git repository `https://github.com/mbround18/valheim-docker.git`
  Installing odin v1.1.0 (https://github.com/mbround18/valheim-docker.git?branch=main#d69ba7fb)
    Updating crates.io index
  Downloaded boxfnonce v0.1.1
  Downloaded clap v3.0.0-beta.2
  Downloaded serde_json v1.0.62
  Downloaded termcolor v1.1.2
  Downloaded strsim v0.10.0
  Downloaded sysinfo v0.16.3
  Downloaded tar v0.4.32
  Downloaded syn v1.0.60
  Downloaded textwrap v0.12.1
  Downloaded xattr v0.2.2
  Downloaded version_check v0.9.2
  Downloaded unicode-segmentation v1.7.1
  Downloaded vec_map v0.8.2
  Downloaded ryu v1.0.5
  Downloaded serde_derive v1.0.123
  Downloaded yaml-rust v0.4.5
  Downloaded once_cell v1.5.2
  Downloaded proc-macro-error v1.0.4
  Downloaded quote v1.0.9
  Downloaded proc-macro-error-attr v1.0.4
  Downloaded linked-hash-map v0.5.4
  Downloaded which v4.0.2
  Downloaded thiserror-impl v1.0.23
  Downloaded proc-macro2 v1.0.24
  Downloaded log v0.4.14
  Downloaded lazy_static v1.4.0
  Downloaded daemonize v0.4.1
  Downloaded libc v0.2.86
  Downloaded os_str_bytes v2.4.0
  Downloaded hashbrown v0.9.1
  Downloaded indexmap v1.6.1
  Downloaded heck v0.3.2
  Downloaded miniz_oxide v0.4.3
  Downloaded itoa v0.4.7
  Downloaded adler v0.2.3
  Downloaded unicode-xid v0.2.1
  Downloaded unicode-width v0.1.8
  Downloaded thiserror v1.0.23
  Downloaded cfg-if v1.0.0
  Downloaded serde v1.0.123
  Downloaded doc-comment v0.3.3
  Downloaded clap_derive v3.0.0-beta.2
  Downloaded flate2 v1.0.20
  Downloaded crc32fast v1.2.1
  Downloaded filetime v0.2.14
  Downloaded autocfg v1.0.1
  Downloaded bitflags v1.2.1
  Downloaded atty v0.2.14
  Downloaded 48 crates (2.3 MB) in 0.36s
   Compiling proc-macro2 v1.0.24
   Compiling unicode-xid v0.2.1
   Compiling libc v0.2.86
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: could not compile `proc-macro2`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: failed to compile `odin v1.1.0 (https://github.com/mbround18/valheim-docker.git?branch=main#d69ba7fb)`, intermediate artifacts can be found at `/tmp/cargo-install2Rg1VG`

**Comment by mbround18** [comment, marked as answer]
I am unsure what you are trying to achieve :P you should install build essentials see my comments on your issue.

**Reply by TheDakk** [reply]
I'm trying to get Odin to work lol,  when I tried to use "odin start" before it kept erroring out with "Error, unable to resolve user name to user id". I restarted my build and ran everything under the docker user account this time.

**Reply by TheDakk** [reply]
Having the same issue, must be missing a config somewhere or I'm confused. I thought may be I needed to install it from outside the container to start and stop the server without having to stop the docker container, but may not even be necessary.

**Reply by mbround18** [reply]
you need to be running odin under a local user <3 i need to update odins instructions <3

**Reply by TheDakk** [reply]
Ahh thanks, also seems like it's not a necessity.
